lib/rust/automata/src/state.rs

```diff
@@ -30,6 +29,13 @@ pub struct State<T> {
     id: usize,
 }
 
+// See https://github.com/mcarton/rust-derivative/issues/112.
```

Comment:
This is sooo bad :( Instead, we might want to disable this lint for now. Adding explicit code ike that makes the code harder to maintain and harder to fix when the lint gets fixed.

Reply:
I will revert this change, as it is a bug in derivative.